python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -47,12 +54,29 @@ def __init__(self):
         self.solid_def = None
         self.populated_by_papermill = False
         self.marshal_dir = None
+        self.info = None
 
     def declare_as_solid(self, repository_def, solid_def_name):
         self.repository_def = repository_def
         self.solid_def = self.repository_def.solid_def_named(solid_def_name)
         self.solid_def_name = check.str_param(solid_def_name, 'solid_def_name')
 
+    def define_out_of_pipeline_info(self, context_config):
+        check.str_param(self.solid_def_name, 'solid_def_name')
+        solid = Solid(self.solid_def_name, self.solid_def)
+        pipeline_def = PipelineDefinition([self.solid_def], name="Ephemeral Notebook Pipeline")
```

Comment:
Because this is in the use case when the notebook is executing as a standalone noteboook (and not in the context of a pipeline run), there is no run_id to inject?